src/flb_output.c

```diff
@@ -208,6 +312,14 @@ int flb_output_task_flush(struct flb_task *task,
                         sizeof(struct flb_output_flush*));
         if (ret == -1) {
             flb_errno();
+            flb_output_flush_destroy(out_flush);
```

Comment:
this was a missing error handling code right?

Reply:
It does look like `flb_task_users_dec(...)` should be called to decrease the task counter. It is harder to tell if we need to call `flb_output_flush_destroy(...)` without working out a call map to see when/where this would be cleaned up under normal/exceptional cases.

Reply:
Here's where it is normally called which is in turn called each time a task finishes.

https://github.com/fluent/fluent-bit/blob/d7a30d1b5ed1c6136f4c4f4169d92e262d87e8da/src/flb_output.c#L501-L534

Since this task is not started it will never finish, meaning that this flush would never get cleaned up unless we call flb_output_flush_destroy here.